src/store/pageview.ts

```diff
@@ -17,6 +26,29 @@ class ViewStore {
     storage.set(StorageKeys.PAGE_VIEWS, this.pageViews);
   }
 
+  _trim() {
+    this.pageViews.sort((a: PageView, b: PageView): number => b.ts - a.ts);
+    this.pageViews = this.pageViews.filter(
+      (pageView: PageView, i: number) =>
+        pageView.ts > timeStampInSecs() - this.maxAge && i < this.storageSize
+    );
+  }
+
+  setMaxAge(maxAge?: number) {
+    if (!maxAge || maxAge < 0 || maxAge === this.maxAge) return;
```

Comment:
What is the thinking with these checks? What are we trying to avoid happening?

Reply:
Those filters invalid inputs and make subsequent setter calls a `no-op` for equal inputs. (sets and trims on the first call, bails out early while the configuration is not changed).
It is a guard clause.

It actually makes the `_trim` method gets called just once per pageView insertion.

Reply:
I understand the reason you added, sorry my question was bad, what I mean is why did you add exactly these **specific** checks?

BTW `!maxAge` make perfects sense, due to how you implemented the config API.

However why would someone pass in the `maxage` twice? Or why would they pass `maxage < 0`?

I just could not understand why those specific checks. You could keep adding stuff there like `|| maxAge > Infinity` etc..

Reply:
Well, `maxAge > Infinity` is not valid input, I think, as it is not possible to come out with a `value > Infinity`.
An `int < 0` is a possible input, tho, yet invalid for our case.
The `maxAge === this.maxAge` makes the method a `no-op` for every page visit after the first one, given that the input will not change. It was there to prevent the calling of the `_trim` method every call. I've removed the _trim from there, tho, so now I can take it out too.